ray-operator/controllers/ray/common/pod.go

```diff
@@ -192,6 +192,11 @@ func DefaultHeadPodTemplate(ctx context.Context, instance rayv1.RayCluster, head
 		// Merge the user overrides from autoscalerOptions into the autoscaler container config.
 		mergeAutoscalerOverrides(&autoscalerContainer, instance.Spec.AutoscalerOptions)
 		podTemplate.Spec.Containers = append(podTemplate.Spec.Containers, autoscalerContainer)
+
```

Comment:
One possible follow up is to unify everything related to Autoscaler into a single place like https://github.com/ray-project/kuberay/pull/2755.